### ISSUE TITLE
Test script failed

### ISSUE THREAD

**Issue by vtereshkov** [shrubb/box-convolutions#23]
Test script assertion failed:

Random seed is 1546545757
Testing for device 'cpu'
Running test_integral_image()...
100%|| 50/50 [00:00<00:00, 1491.15it/s]
OK
Running test_box_convolution_module()...
  0%|
python3: /pytorch/third_party/ideep/mkl-dnn/src/cpu/jit_avx2_conv_kernel_f32.cpp:567: static mkldnn::impl::status_t mkldnn::impl::cpu::jit_avx2_conv_fwd_kernel_f32::init_conf(mkldnn::impl::cpu::jit_conv_conf_t&, const convolution_desc_t&, const mkldnn::impl::memory_desc_wrapper&, const mkldnn::impl::memory_desc_wrapper&, const mkldnn::impl::memory_desc_wrapper&, const primitive_attr_t&): Assertion `jcp.ur_w * (jcp.nb_oc_blocking + 1) <= num_avail_regs' failed.
Aborted (core dumped)

Configuration: Ubuntu 16.04 LTS, CUDA 9.2, PyTorch 1.1.0, GCC 5.4.0.

**Comment by shrubb** [comment]
This is a PyTorch / MKL-DNN bug: pytorch/pytorch#20583. I suggest that you follow that link and try running the reproducing script from there. If it fails too, try installing a nightly PyTorch build. If that in turn doesn't help, report to PyTorch team via that issue.

**Comment by vtereshkov** [comment]
I've reproduced the PyTorch bug. Now looking for a Python 3 version of the nightly build.

**Comment by vtereshkov** [comment]
The nightly build still has this problem. Downgraded to PyTorch 1.0.0 by running

pip3 uninstall torch
pip3 uninstall torchvision
pip3 install https://download.pytorch.org/whl/cu90/torch-1.0.0-cp35-cp35m-linux_x86_64.whl
pip3 install --no-deps torchvision==0.2.1

Now the test is passed.

**Comment by shrubb** [comment]
Glad it worked. Thanks for investigating.

Now I understand that the test fails because in it I use `nn.Conv2d` with unusual parameters that induce a bug in MKL-DNN. So, despite the test failing, I think you can still safely use `box-convolutions` on PyTorch 1.1.0.
